### DISCUSSION TITLE
Alarm_control_panel

### DISCUSSION THREAD

**Discussion by calz64** [indomus/forum#620] [HUB: Home Assistant]
Ciao a tutti, mi sapete dire che errore è questo. Il pannello di allarme, e la sua entità, funzionano per un po', poi dopo qualche riavvio di HA, allarm_control_panel scompare e mi da questo errore.

```
Logger: homeassistant.components.alarm_control_panel
Source: components/manual/alarm_control_panel.py:422
Integration: Pannello di Controllo degli Allarmi (documentation, issues)
First occurred: 11:34:47 (2 occurrences)
Last logged: 11:34:47

Error adding entities for domain alarm_control_panel with platform manual
Error while setting up manual platform for alarm_control_panel
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 587, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 710, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/usr/src/homeassistant/homeassistant/components/manual/alarm_control_panel.py", line 422, in async_added_to_hass
    and state.attributes[ATTR_PREVIOUS_STATE]
KeyError: 'previous_state'
```

In rete suggeriscono di cambiare nome ad alarm e cosi al riavvio ricompare, E' vero ho provato, ma così saltano le automazioni fatte.

**Comment by calz64** [comment]
Forse la soluzione è questa
https://github.com/gazoscalvertos/HASS-Persistence
mi suggerite di installare il repo come spiegato, non capisco bene cosa fa, è ok?

**Reply by snakuzzo** [reply]
L'ultima volta che lo sviluppatore ha mesos mano a quel custom component risale a 4 anni fa.
Con le opportune modifiche (visto che lato HA qualcosa è cambiato nel tempo) magari funziona, ma che c'entra con il tuo problema ?
Questo custom (così dice) dovrebbe rendere persistente lo stato di una entità che definisci in configurazione.
Tu invece dici che sparisce proprio l' `alarm_control_panel` ...non è che si perde lo stato.

Com'è fatto questo alarm control panel ? Cos'ha di particolare ?

**Reply by calz64** [reply]
Grazie della spiegazione, in effetti non saprei.
La configurazione è questa, ma cosa significa 
KeyError: 'previous_state'
```
# Pannello per allarme casa
alarm_control_panel:
  - platform: manual
    name: casa
    code: '....'
    arming_time: 30
    delay_time: 20
    trigger_time: 4
    disarmed:
      trigger_time: 0
    armed_home:
      arming_time: 0
      delay_time: 0
```

**Reply by snakuzzo** [reply]
c'è una [issue](https://github.com/home-assistant/core/issues/51428) di inizio giugno sul progetto di Home Assistant  ed è ancora in stato aperta. Seguila...e magari votala pure con il "pollice" (non rispondendo)
